File: eratosthenes.go
package main

import "fmt"

type Run interface {
	 run() int
}

type Math struct {

	 n int	 
}

func (m Math) run() int {

	  totalPrimes := m.n - 1

	  len := m.n

	  Primes := make([]int, len+1)

	  for i:=2; i < len + 1; i++ {

	  	  Primes[i] = 1
	  }

	  for i := 2; i * i <= len; i++ {

	  	  if Primes[i] == 1 {

             j := 2

             for i * j <= len {

                   multiply := i * j

                   if Primes[multiply] == 1 {

                   	  totalPrimes--
                   }   
                   Primes[multiply] = 0 
                   j++;
             } 
	  	  }

	  }

  	  for i:=2; i < len + 1; i++ {

  	  	     if(Primes[i] == 1) {
  	  	     	 fmt.Printf("%d ", i)
  	  	     } 	 
  	  }	

      fmt.Printf("\n%s -> ", "count")
	  return totalPrimes
}

func main() {
          
     eratosthene := Math{7920}

     fmt.Println(eratosthene.run())
}

File: eratosthenes.rb
class Eratosthenes

	  def initialize(n)

	  	  @totalPrimes = n - 1

          _run n          

	  end	

	  def _run n

	  	  primes = Array.new( n + 1, 1 )	  	  
          
          for i in 2..n          	  
          	  if i * i >= n
          	  	 break
          	  end	

            if primes[ i ] == 1
             	              
          	  j = 2

          	  while i * j <= n
                   
          	  	 multiply = i * j

          	  	 if primes[ multiply ] == 1
          	  	    @totalPrimes -= 1
          	  	 end 	

          	  	 primes[ multiply ] = 0

          	  	 j += 1

          	  end

          	end         	  
          end	

          
          for i in 2..n
          	  if primes[i] == 1
          	  	 print i, " "
          	  end	
          end
	  end	

	  def getCounts
	  	  return @totalPrimes
	  end	
end	

ob = Eratosthenes.new(7920)
printf("\n%d\n", ob.getCounts())